app/classifier/drawing-tools/curve.cjsx

```diff
@@ -0,0 +1,191 @@
+React = require 'react'
+DrawingToolRoot = require './root'
+DragHandle = require './drag-handle'
+Draggable = require '../../lib/draggable'
+DeleteButton = require './delete-button'
+
+FINISHER_RADIUS = 8
+GRAB_STROKE_WIDTH = 6
+GUIDE_WIDTH = 1
+GUIDE_DASH = [4, 4]
+# fraction of line lenght along (x) and perpendicular (y) to the line to place control point
+DEFAULT_CURVE = {x: 0.5, y: 0}
+
+DELETE_BUTTON_WEIGHT = 2 # Weight of the second point.
+
+module.exports = React.createClass
+  displayName: 'CurveTool'
+
+  statics:
+    initCoords: null
+
+    defaultValues: ({x, y}) ->
+      points: []
+      closed: false
+
+    initStart: (newPoint, mark) ->
+      if mark.points.length > 0
+        lastEnd = mark.points[-1..][0]
+        cp = @toImageFrame(lastEnd, newPoint, DEFAULT_CURVE)
```

Comment:
I'm not sure what `cp` stands for. Let's try not to abbreviate anything, the more descriptive the better.